optimum/onnxruntime/modeling_decoder.py

```diff
@@ -0,0 +1,516 @@
+            )
+
+        return cls(*model, config=config, **kwargs)
+
+    @classmethod
+    def _from_transformers(
+        cls,
+        model_id: str,
+        save_dir: Union[str, Path] = default_cache_path,
+        use_auth_token: Optional[Union[bool, str, None]] = None,
+        revision: Optional[Union[str, None]] = None,
+        force_download: bool = True,
+        cache_dir: Optional[str] = None,
+        **kwargs,
+    ):
+        """
+        Exports through the ONNX format a vanilla Transformers model using `transformers.onnx.export_onnx`.
+
+        Arguments:
+            model_id (`str` or `Path`):
+                The directory from which to load the model.
+            save_dir (`str` or `Path`):
+                The directory where the ONNX model should be saved, default to
+                `transformers.file_utils.default_cache_path`, which is the cache dir for transformers.
+            use_auth_token (`str` or `bool`):
+                The token to use as HTTP bearer authorization for remote files. Needed to load models from a private
+                repository.
+            revision (`str`):
+                The specific model version to use. It can be a branch name, a tag name, or a commit id.
+            force_download (`bool`, *optional*, defaults to `False`):
+                Whether or not to force the (re-)download of the model weights and configuration files, overriding the
+                cached versions if they exist.
+            cache_dir (`Union[str, Path]`, *optional*):
+                The path to a directory in which a downloaded pretrained model configuration should be cached if the
+                standard cache should not be used.
+            use_cache (`bool`, *optional*, defaults to `True`):
+                Whether or not use the pre-computed key/values hidden-states in order to speed up sequential decoding.
+            kwargs (`Dict`, *optional*):
+                kwargs will be passed to the model during initialization.
+        """
+        # Create local save dir in cache dir
+        save_dir = Path(save_dir).joinpath(model_id)
+        save_dir.mkdir(parents=True, exist_ok=True)
+        kwargs["model_save_dir"] = save_dir
+        # use_cache = kwargs.get("use_cache", None)
+        use_cache = kwargs.get("use_cache", True)
+
+        preprocessor = get_preprocessor(model_id)
+        model = FeaturesManager.get_model_from_feature(cls.pipeline_task, model_id)
+        # use_cache = use_cache if use_cache is not None else model.config.use_cache
+        _, model_onnx_config = FeaturesManager.check_supported_model_or_raise(model, feature=cls.pipeline_task)
+        onnx_config = model_onnx_config(model.config)
+        onnx_opset = onnx_config.default_onnx_opset
+
+        # Export the decoder without the past key values
+        onnx_config = DecoderOnnxConfigWithPast(model.config, task=cls.pipeline_task, use_past=False)
```

Comment:
I think we can remove un-use code

```suggestion
        # Export the decoder without the past key values
        onnx_config = DecoderOnnxConfigWithPast(model.config, task=cls.pipeline_task, use_past=False)
        onnx_opset = onnx_config.default_onnx_opset
```

Reply:
Hi @tiena2cva, thanks for the heads up. Unused codes are removed in the refactoring.